specs/waku/v2/waku-store.md

```diff
@@ -8,23 +8,43 @@ authors: Oskar Thorén <oskar@status.im>, Dean Eigenmann <dean@status.im>, Sanaz
 # Table of Contents
 
 - [Abstract](#abstract)
+  - [Security Requirements](#security-requirements)
+    - [Terminologies](#terminologies)
+  - [Adversarial Model](#adversarial-model)
 - [Wire Specification](#wire-specification)
   - [Protobuf](#protobuf)
     - [Index](#index)
     - [PagingInfo](#paginginfo)
     - [HistoryQuery](#historyquery)
     - [HistoryResponse](#historyresponse)
+  - [Security Analysis](#security-analysis)
+  - [Future Work](#future-work)
 - [Changelog](#changelog)
     - [2.0.0-beta2](#200-beta2)
     - [2.0.0-beta1](#200-beta1)
 - [Copyright](#copyright)
 
 # Abstract
 
-This specification explains the Waku Store protocol which enables querying of messages received through relay protocol and stored by other nodes. It also supports pagination for more efficient querying of historical messages. 
+This specification explains the Waku `store` protocol which enables querying of messages received through relay protocol and stored by other nodes. It also supports pagination for more efficient querying of historical messages. 
 
 **Protocol identifier***: `/vac/waku/store/2.0.0-beta2`
 
+## Security Requirements
+
+- **Anonymous query**: This feature guarantees that nodes can anonymously query historical messages from other nodes (i.e., without disclosing the exact topics of waku messages they are interested in).  As such, no adversary in the `store` protocol would be able to learn which peer is interested in which topics of waku message.
```

Comment:
Is this accurate? It seems to me that disclosing topic interest is the trade-off used for store protocol. Could you elaborate?

Reply:
Different approaches to achieve anonymous query is discussed in the "Future work" section, please let me whether more clarification is required.